sentry-spring-boot-starter/src/main/java/io/sentry/spring/boot/SentrySecurityFilter.java

```diff
@@ -0,0 +1,43 @@
+package io.sentry.spring.boot;
+
+import com.jakewharton.nopen.annotation.Open;
+import io.sentry.core.IHub;
+import java.io.IOException;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+/**
+ * Adds {@link SentryUserHttpServletRequestProcessor} to the scope in order to decorate {@link
+ * io.sentry.core.SentryEvent} with principal name and user ip address.
+ */
+@Open
+public class SentrySecurityFilter extends OncePerRequestFilter {
+  private final @NotNull IHub hub;
+
+  public SentrySecurityFilter(final @NotNull IHub hub) {
+    this.hub = hub;
+  }
+
+  @Override
+  protected void doFilterInternal(
+      final @NotNull HttpServletRequest request,
+      final @NotNull HttpServletResponse response,
+      final @NotNull FilterChain filterChain)
+      throws ServletException, IOException {
+    hub.configureScope(
+        scope ->
+            scope.addEventProcessor(
+                new SentryUserHttpServletRequestProcessor(
+                    request.getUserPrincipal(), toIpAddress(request))));
+    filterChain.doFilter(request, response);
+  }
+
+  private static @NotNull String toIpAddress(final @NotNull HttpServletRequest request) {
+    final String ipAddress = request.getHeader("X-FORWARDED-FOR");
```

Comment:
Should this be opt-in?
Would technically expose the server to report a fake client IP if the request was sent to the server by the client with this (no-proxy) set?

Also, I believe the value here can be a list (chained-proxy servers) so might require some more handling.